### ISSUE TITLE
NullPointerException on commit.author.login

### ISSUE THREAD

**Issue by brioche-cooperl** [requarks/changelog-action#20]
Hello,

I'm encountering an error when trying to generate a changelog on a repo. The action log doesn't show exactly which commit is concerned, but nothing caught my eye in the git log.

```
/home/runner/work/_actions/requarks/changelog-action/v1/dist/index.js:28278
          author: commit.author.login,
                                ^

TypeError: Cannot read properties of null (reading 'login')
    at main (/home/runner/work/_actions/requarks/changelog-action/v1/dist/index.js:28278:33)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Sorry I can't show you the repo we're using this on since it's private.

**Comment by jmartinak** [comment]
@brioche-cooperl Had the same problem. The problem is that your commits aren't linked to your GH profile via `user.email` git config. Read this https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user

The action implementation relies on the `commit.author` but this is `null` in such cases. 

Just make sure the commits contain the correct email https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

**Comment by NGPixel** [comment]
Fixed in v1.7.0. However, as @jmartinak mentioned, you need to use a valid email address with your commits that match a GitHub account, otherwise no reference to the authors will be added.
